### PR TITLE
fix one potential crash issue when directional intra modes enabled

### PR DESCRIPTION
**Pull request by rainliu**
when directional intra modes are enabled, running "target\release\rav1e.exe nyan.y4m -o enc_file.ivf --limit 1 -s 3 -r rec_file.y4m" will cause crash due to hard coded ArrayVec::<_;7>.

This commit fixed this potential crash issue.

### CODE REVIEW

**Comment by lu-zero** [comment]
I would reword the commit message a little, but sounds a good idea to set the array to the upper bound to avoid surprises later. @tdaede do you agree?

**Comment by tdaede** [comment]
This change is correct, but can we use .len() (is it a const fn?)

**Comment by rainliu** [comment]
@tdaede, .len() is not const fn. 

I am trying to simplify this change by using PredictionMode::NEARESTMV as size of ArrayVec:

```
-let mut modes = ArrayVec::<[_;RAV1E_INTRA_MODES_LEN]>::new();
+let mut modes = ArrayVec::<[_;PredictionMode::NEARESTMV as usize]>::new();
```

**Comment by tdaede** [comment]
I think the way the patch is currently written is more robust than doing that. Another constant you could use is INTRA_MODES - it's higher than the number of intra modes we currently test, but wasting a bit of space here isn't an issue.

**Comment by yuryanliu** [comment]
just updated commit by using INTRA_MODES
